Update container-runtimes.md

esbc-disciple opened the pull request:
Added a clarification on setting systemd as the cgroup driver. Many folks setting up kubernetes for the first time are not going to understand what a crash loop is or if they're having one.

There is a lot I don't know, so maybe it's not always the case that your config.toml should already be populated. But... if it's acceptable to run the `containerd config defaults` command, it makes sense to suggest that from the start to keep people from wasting time when their kubeapi isn't up after the `kubeadm init` command.

kundan2707 reviewed (commented):
these information seems helpful
thanks

kundan2707 commented:
/lgtm

neolit123 commented on content/en/docs/setup/production-environment/container-runtimes.md:
```diff
@@ -233,6 +233,11 @@ To use the `systemd` cgroup driver in `/etc/containerd/config.toml` with `runc`,
 The `systemd` cgroup driver is recommended if you use [cgroup v2](/docs/concepts/architecture/cgroups).
 
 {{< note >}}
+The elipses above indicate that config.toml should already have these sections. If you
```
did you mean "ellipsis" instead of "elipses"?
i would avoid the word either way, and use something more common.

but....instead of amending this note which is about plugins, how about adding a new note on top of the example TOML saying in the lines of "run command FOO  if your TOML file does not exist yet"?

sftim replied:
elipses is strictly correct, but it's an obscure word.

neolit123 commented:
/sig node

esbc-disciple commented:
I installed containerd from a Debian package via apt. When you do that, the config.toml file gets created but has almost nothing in it (except for the line that disables cri, which you have to remove per the Kubernetes container runtime docs).

I originally kept the (bad) toml that came with the containerd install, removed the line that disabled cri, and added the toml snippet from this doc into it. It wasn’t until I ran the containerd config default command that I realized that there is likely much more that needs to be in that toml file. Simply adding those lines from the toml snippet isn’t enough.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Lubomir I. Ivanov ***@***.***>
Sent: Wednesday, December 13, 2023 5:11:12 AM
To: kubernetes/website ***@***.***>
Cc: esbc-disciple ***@***.***>; Mention ***@***.***>
Subject: Re: [kubernetes/website] Update container-runtimes.md (PR #44313)


/sig node

—
Reply to this email directly, view it on GitHub<https://github.com/kubernetes/website/pull/44313#issuecomment-1853717540>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AXAQ2JUB4YF6JCBRDOSWGRLYJGENBAVCNFSM6AAAAABASEZPCSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNJTG4YTONJUGA>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

neolit123 commented:
> I installed containerd from a Debian package via apt. When you do that, the config.toml file gets created but has almost nothing in it (except for the line that disables cri, which you have to remove per the Kubernetes container runtime docs). I originally kept the (bad) toml that came with the containerd install, removed the line that disabled cri, and added the toml snippet from this doc into it. It wasn’t until I ran the containerd config default command that I realized that there is likely much more that needs to be in that toml file. Simply adding those lines from the toml snippet isn’t enough.


i don't think we want to include minimal TOML or exact steps on what to modify.
if the containerd config is in flux, or different between distros we don't want to maintain such config variations at k8s.io.
we just want users to set the cgroup driver part.

that said, if you have an example config in mind that is stable and portable, please add it in the PR and provide some notes on what it does. we could also link to the containerd documentation as it's a trusted project.

esbc-disciple commented:
I may be missing something, but it doesn't seem like you need to hard-code anything in the docs other than what is already there.

The intent of my proposed change was to hopefully make it easier to determine when you need to reset the containerd config (already an accepted step in the docs, below). Maybe instead of my PR, y'all can just add a parenthetical after ... crash loops...

Something like:
If you experience container crash loops (or get an error message like "The connection to the server localhost:6443 was refused")... consider resetting the configuration..."

"If you experience crash loops" and "if you get a [localhost:6443 connection error]" are essentially saying the same thing. I just think the latter is more meaningful to complete novices like me. After research, I learned that I was getting that error because kubeapi pod was not starting. However, it took some time to connect that to the "crash loop" verbiage in the docs.

Current kubernetes.io cgroup driver verbiage:
If you experience container crash loops after the initial cluster installation or after installing a CNI, the containerd configuration provided with the package might contain incompatible configuration parameters. Consider resetting the containerd configuration with containerd config default > /etc/containerd/config.toml as specified in getting-started.md<https://github.com/containerd/containerd/blob/main/docs/getting-started.md#advanced-topics> and then set the configuration parameters specified above accordingly.


Sent from Outlook<http://aka.ms/weboutlook>

________________________________
From: Lubomir I. Ivanov ***@***.***>
Sent: Wednesday, December 13, 2023 9:18 AM
To: kubernetes/website ***@***.***>
Cc: esbc-disciple ***@***.***>; Mention ***@***.***>
Subject: Re: [kubernetes/website] Update container-runtimes.md (PR #44313)


I installed containerd from a Debian package via apt. When you do that, the config.toml file gets created but has almost nothing in it (except for the line that disables cri, which you have to remove per the Kubernetes container runtime docs). I originally kept the (bad) toml that came with the containerd install, removed the line that disabled cri, and added the toml snippet from this doc into it. It wasn’t until I ran the containerd config default command that I realized that there is likely much more that needs to be in that toml file. Simply adding those lines from the toml snippet isn’t enough.

i don't think we want to include minimal TOML or exact steps on what to modify.
if the containerd config is in flux, or different between distros we don't want to maintain such config variations at k8s.io.
we just want users to set the cgroup driver part.

that said, if you have an example config in mind that is stable and portable, please add it in the PR and provide some notes on what it does. we could also link to the containerd documentation as it's a trusted project.

—
Reply to this email directly, view it on GitHub<https://github.com/kubernetes/website/pull/44313#issuecomment-1854122565>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AXAQ2JUQ2IROCP474A3BXO3YJHBNTAVCNFSM6AAAAABASEZPCSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQNJUGEZDENJWGU>.
You are receiving this because you were mentioned.

neolit123 commented:
i do not disagree with the current state of this PR, but requested a change to add a separate note:
https://github.com/kubernetes/website/pull/44313/files#r1425203432

divya-mohan0209 commented:
Explicit hold to avoid accidental merges till comments by @neolit123 have been addressed. 
This will be followed by an official docs review.
/hold

divya-mohan0209 commented:
@esbc-disciple : Please could you advise if you're okay making the changes mentioned by @neolit123? Thanks!

kundan2707 commented:
/remove-lgtm

esbc-disciple commented:
I agree with the changes suggested by @nolit123.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Kundan Kumar ***@***.***>
Sent: Sunday, January 21, 2024 10:00:16 PM
To: kubernetes/website ***@***.***>
Cc: esbc-disciple ***@***.***>; Mention ***@***.***>
Subject: Re: [kubernetes/website] Update container-runtimes.md (PR #44313)


/remove-lgtm

—
Reply to this email directly, view it on GitHub<https://github.com/kubernetes/website/pull/44313#issuecomment-1903163137>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AXAQ2JW35IKCBI7Z3NITXTDYPXP5BAVCNFSM6AAAAABASEZPCSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMBTGE3DGMJTG4>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

esbc-disciple commented:
/remove-lifecycle rotten

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Kubernetes Triage Robot ***@***.***>
Sent: Tuesday, May 21, 2024 7:24:48 AM
To: kubernetes/website ***@***.***>
Cc: esbc-disciple ***@***.***>; Mention ***@***.***>
Subject: Re: [kubernetes/website] Update container-runtimes.md (PR #44313)


The Kubernetes project currently lacks enough active contributors to adequately respond to all PRs.

This bot triages PRs according to the following rules:

  *   After 90d of inactivity, lifecycle/stale is applied
  *   After 30d of inactivity since lifecycle/stale was applied, lifecycle/rotten is applied
  *   After 30d of inactivity since lifecycle/rotten was applied, the PR is closed

You can:

  *   Mark this PR as fresh with /remove-lifecycle rotten
  *   Close this PR with /close
  *   Offer to help out with Issue Triage<https://www.kubernetes.dev/docs/guide/issue-triage/>

Please send feedback to sig-contributor-experience at kubernetes/community<https://github.com/kubernetes/community>.

/lifecycle rotten

—
Reply to this email directly, view it on GitHub<https://github.com/kubernetes/website/pull/44313#issuecomment-2122517456>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AXAQ2JXRCF7LKFMYL4EW2MTZDM4JBAVCNFSM6AAAAABASEZPCSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCMRSGUYTONBVGY>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

esbc-disciple commented:
I'm sorry. Is this waiting on me? The approval process has me a little turned around. Do I just need to re-commit the changes we discussed?

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Aaron Miller ***@***.***>
Sent: Tuesday, May 21, 2024 7:32:33 AM
To: kubernetes/website ***@***.***>; kubernetes/website ***@***.***>
Cc: Mention ***@***.***>
Subject: Re: [kubernetes/website] Update container-runtimes.md (PR #44313)

/remove-lifecycle rotten

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Kubernetes Triage Robot ***@***.***>
Sent: Tuesday, May 21, 2024 7:24:48 AM
To: kubernetes/website ***@***.***>
Cc: esbc-disciple ***@***.***>; Mention ***@***.***>
Subject: Re: [kubernetes/website] Update container-runtimes.md (PR #44313)


The Kubernetes project currently lacks enough active contributors to adequately respond to all PRs.

This bot triages PRs according to the following rules:

  *   After 90d of inactivity, lifecycle/stale is applied
  *   After 30d of inactivity since lifecycle/stale was applied, lifecycle/rotten is applied
  *   After 30d of inactivity since lifecycle/rotten was applied, the PR is closed

You can:

  *   Mark this PR as fresh with /remove-lifecycle rotten
  *   Close this PR with /close
  *   Offer to help out with Issue Triage<https://www.kubernetes.dev/docs/guide/issue-triage/>

Please send feedback to sig-contributor-experience at kubernetes/community<https://github.com/kubernetes/community>.

/lifecycle rotten

—
Reply to this email directly, view it on GitHub<https://github.com/kubernetes/website/pull/44313#issuecomment-2122517456>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AXAQ2JXRCF7LKFMYL4EW2MTZDM4JBAVCNFSM6AAAAABASEZPCSVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCMRSGUYTONBVGY>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

divya-mohan0209 commented:
@esbc-disciple : Yep, it's awaiting changes from your side. @neolit123 has added his comments in [this review](https://github.com/kubernetes/website/pull/44313#issuecomment-1854344272)

esbc-disciple reviewed (commented):
I agree with proposed changes. It fulfills the spirit of why I started this pull request but is much cleaner.

neolit123 commented:
TOML sections seems clear enough
in INI they are called sections
